logicrunner/handle_calls.go

```diff
@@ -153,3 +176,91 @@ func (h *HandleCall) Present(ctx context.Context, f flow.Flow) error {
 	return nil
 
 }
+
+type HandleAdditionalCallFromPreviousExecutor struct {
+	dep *Dependencies
+
+	Message bus.Message
+}
+
+// This is basically a simplified version of HandleCall.handleActual().
+// Please note that currently we lack any fraud detection here.
+// Ideally we should check that the previous executor was really an executor
+// during previous pulse, that the request was really registered, etc.
+// Also we don't handle case when pulse changes during execution of this handle.
+// In this scenario user is in a bad luck. The request will be lost and user will have
+// to re-send it after some timeout.
+func (h *HandleAdditionalCallFromPreviousExecutor) handleActual(
+	ctx context.Context,
+	msg *message.AdditionalCallFromPreviousExecutor,
+	f flow.Flow,
+) {
+	lr := h.dep.lr
+	ref := msg.ObjectReference
+
+	os := lr.UpsertObjectState(ref)
+
+	os.Lock()
+	if os.ExecutionState == nil {
+		os.ExecutionState = &ExecutionState{
+			Ref:   ref,
+			Queue: make([]ExecutionQueueElement, 0),
+		}
+	}
+	es := os.ExecutionState
+	os.Unlock()
+
+	es.Lock()
+	qElement := ExecutionQueueElement{
+		ctx:     ctx,
```

Comment:
Hmmm.. I don't think so. This is a context passed to a handler, it contains Jaeger trace ids and everything inherited from original VE. I would like to keep all this info unless we have a good reason to do otherwise.